yt_dlp/extractor/safari.py

```diff
@@ -101,39 +101,36 @@ class SafariIE(SafariBaseIE):
     _UICONF_ID = '29375172'
 
     def _real_extract(self, url):
+        query = {}
         mobj = self._match_valid_url(url)
 
         reference_id = mobj.group('reference_id')
         if reference_id:
-            video_id = reference_id
-            partner_id = self._PARTNER_ID
-            ui_id = self._UICONF_ID
+            query['flashvars[referenceId]'] = video_id = reference_id
+            query['wid'] = '_%s' % self._PARTNER_ID
+            query['uiconf_id'] = self._UICONF_ID
         else:
             video_id = '%s-%s' % (mobj.group('course_id'), mobj.group('part'))
 
             webpage, urlh = self._download_webpage_handle(url, video_id)
 
             mobj = re.match(self._VALID_URL, urlh.geturl())
             reference_id = mobj.group('reference_id')
-            if not reference_id:
-                reference_id = self._search_regex(
-                    r'data-reference-id=(["\'])(?P<id>(?:(?!\1).)+)\1',
-                    webpage, 'kaltura reference id', group='id')
-            partner_id = self._search_regex(
+            if reference_id:
+                query['flashvars[referenceId]'] = reference_id
+            else:
+                query['entry_id'] = self._search_regex(
```

Comment:
If `reference_id` can't be determined neither from `url` or `urlh` (and `webpage` no longer has it), `entry_id` is used instead.